src/Control/Director.php

```diff
@@ -76,40 +80,47 @@ class Director implements TemplateGlobalProvider
     private static $alternate_base_folder;
 
     /**
+     * Setting this explicitly specifies the protocol ("http" or "https") used, overriding the normal
+     * behaviour of Director::is_https introspecting it from the request
+     *
      * @config
      *
-     * @var array
+     * @var string
      */
-    protected static $dev_servers = array();
+    private static $alternate_protocol;
 
     /**
-     * @config
+     * Setting this overrides the host name
      *
-     * @var array
+     * @config
+     * @var string
      */
-    protected static $test_servers = array();
+    private static $alternate_host;
```

Comment:
I think we should remove alternate_host in favour of alternate_base_url.

Note that this doesn't need to be logged as an API change as it was introduced in 4.